src/init_DifferentialEquations.jl

```diff
@@ -1,6 +1,41 @@
 using .DifferentialEquations
 const DE = DifferentialEquations
 
+import .DifferentialEquations: controls
+
+export trajectories,
+       inputs
+
+struct SimulationSolution{TT, CT, IT}
+    trajectory::TT  # trajectory pieces for each control cycle
+    controls::CT  # control inputs for each control cycle
+    inputs::IT  # nondeterministic inputs for each control cycle
+end
+
+trajectory(sol::SimulationSolution) = sol.trajectory
+controls(sol::SimulationSolution) = sol.controls
+inputs(sol::SimulationSolution) = sol.inputs
+
+struct EnsembleSimulationSolution{TT, CT, IT}
+    solutions::Vector{SimulationSolution{TT, CT, IT}}
+end
+
+# constructor from a bulk input
+function EnsembleSimulationSolution(simulations, controls, inputs)
+    @assert length(simulations) == length(controls) == length(inputs) "incompatible lengths"
+    solutions = @inbounds [SimulationSolution(simulations[i], controls[i],
+        isassigned(inputs, i) ? inputs[i] : nothing)
+        for i in eachindex(simulations)]
+    return EnsembleSimulationSolution(solutions)
+end
+
+Base.length(ess::EnsembleSimulationSolution) = length(ess.solutions)
+solution(ess::EnsembleSimulationSolution, i) = ess.solutions[i]
+trajectory(ess::EnsembleSimulationSolution, i) = trajectory(solution(ess, i))
+trajectories(ess::EnsembleSimulationSolution) = trajectory.(ess.solutions)
```

Comment:
I am not sure about this name. `trajectories` returns the solutions, not just the trajectories. Should it rather be called `solutions`?

Reply:
does it? i expect `trajectory.(ess.solutions)`  to return a vector of all computed trajectories

Reply:
Oh, you are right :facepalm: